.github/workflows/java-roll-version.yaml

```diff
@@ -0,0 +1,39 @@
+# A workflow to find the current version from pom and then commit the release version (with tag) and the next dev cycle version
+# Note this uses java-java-update-version.yaml
+name: Roll maven version
+on:
+  workflow_call:
+
+jobs:
+  extract:
+    name: extract-version
+    runs-on:
+      - self-hosted
+      - java
+    outputs:
+      major: ${{ steps.major.outputs.major }}
+      minor: ${{ steps.minor.outputs.minor }}
+      patch: ${{ steps.patch.outputs.patch }}
+    steps:
+      - uses: actions/checkout@v4
+      - id: major
+        run: |
+          echo ::set-output name=major::$(mvn build-helper:parse-version help:evaluate -Dexpression=parsedVersion.majorVersion -q -DforceStdout --no-transfer-progress)
+      - id: minor
+        run: |
+          echo ::set-output name=minor::$(mvn build-helper:parse-version help:evaluate -Dexpression=parsedVersion.minorVersion -q -DforceStdout --no-transfer-progress)
+      - id: patch
+        run: |
+          echo ::set-output name=patch::$(mvn build-helper:parse-version help:evaluate -Dexpression=parsedVersion.incrementalVersion -q -DforceStdout --no-transfer-progress)
+  echo-version:
```

Comment:
Same idea here -- there's no need to start a separate job / container, since it just waits for `needs: extra`.